applications/plugins/SofaMatrix/src/SofaMatrix/SimpleImageViewerWidget.h

```diff
@@ -0,0 +1,211 @@
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+#include <SofaMatrix/config.h>
+
+#include <sofa/gui/qt/SimpleDataWidget.h>
+#include <image/image_gui/ImagePlaneWidget.h>
+
+namespace sofa::gui::qt
+{
+
+
+class SimpleImageViewerGraphScene : public QGraphicsScene
+{
+    Q_OBJECT
+public:
+    explicit SimpleImageViewerGraphScene(QImage* im, QObject *parent = nullptr)
+    : QGraphicsScene(parent)
+    , m_image(im)
+    {
+        assert(im);
+        this->setSceneRect(0,0, m_image->width(), m_image->height());
+    }
+
+    void drawBackground(QPainter *painter, const QRectF &rect) override
+    {
+        QGraphicsScene::drawBackground(painter, rect);
+        if(m_image)
+        {
+            painter->drawImage(this->sceneRect(),*m_image);
+        }
+    }
+
+private:
+    const QImage *m_image;
+};
+
+class BaseSimpleImageViewerGraphWidget : public QGraphicsView
+{
+    Q_OBJECT
+
+public:
+    using QGraphicsView::QGraphicsView;
+};
+
+template<class SimpleBitmapType>
+class SimpleImageViewerGraphWidget : public BaseSimpleImageViewerGraphWidget
+{
+public:
+    SimpleImageViewerGraphWidget(QWidget * parent, const SimpleBitmapType& plane)
+    : BaseSimpleImageViewerGraphWidget(parent)
+    , m_imageplane(&plane)
+    {
+        readFromData(plane);
+        const QSizePolicy sizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+        setSizePolicy(sizePolicy);
+
```

Comment:
```suggestion
        QVBoxLayout* vbox = new QVBoxLayout();
        vbox->addStretch();
        this->setLayout(vbox);
        
```

Reply:
Tada! qt layout logic magic!

Reply:
Thanks a lot! You're my Qt expert now! Thanks again